src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/reducers/__tests__/chartBuilderReducer.test.ts

```diff
@@ -966,5 +1152,371 @@ describe('chartBuilderReducer', () => {
+              dataGrouping: { type: 'Custom', customGroups: [] },
+              boundaryLevel: 30,
+            },
+          ],
+        },
+      });
+    });
+    test('setting data groupings does not change boundary levels', () => {
+      const initialConfiguration: Chart = {
+        legend: {
+          position: 'top',
+          items: [],
+        },
+        axes: {
+          major: {
+            type: 'major',
+            dataSets: [
+              {
+                filters: ['ethnicity-major-chinese', 'state-funded-primary'],
+                indicator: 'authorised-absence-sessions',
+                timePeriod: '2015_AY',
+              },
+              {
+                filters: ['ethnicity-major-chinese', 'state-funded-primary'],
+                indicator: 'authorised-absence-sessions',
+                timePeriod: '2014_AY',
+              },
+            ],
+            referenceLines: [],
+          },
+        },
+        type: 'map',
+        height: 300,
+        title: 'Chart title',
+        alt: '',
+        boundaryLevel: 1,
+        map: {
+          dataSetConfigs: [
+            {
+              dataSet: {
+                filters: ['ethnicity-major-chinese', 'state-funded-primary'],
+                indicator: 'authorised-absence-sessions',
+                timePeriod: '2015_AY',
+              },
+              dataGrouping: { type: 'EqualIntervals', customGroups: [] },
+              boundaryLevel: 2,
+            },
+            {
+              dataSet: {
+                filters: ['ethnicity-major-chinese', 'state-funded-primary'],
+                indicator: 'authorised-absence-sessions',
+                timePeriod: '2014_AY',
+              },
+              dataGrouping: { type: 'Custom', customGroups: [] },
+              boundaryLevel: 3,
+            },
+          ],
+        },
+      };
+      const expectedChartBuilderStateProps: Omit<ChartBuilderState, 'map'> = {
+        axes: {
+          major: {
+            type: 'major',
+            groupBy: 'locations',
+            dataSets: [
+              {
+                filters: ['ethnicity-major-chinese', 'state-funded-primary'],
+                indicator: 'authorised-absence-sessions',
+                timePeriod: '2015_AY',
+              },
+              {
+                filters: ['ethnicity-major-chinese', 'state-funded-primary'],
+                indicator: 'authorised-absence-sessions',
+                timePeriod: '2014_AY',
+              },
+            ],
+            referenceLines: [],
+            visible: true,
+            label: {
+              text: '',
+            },
+          },
+        },
+        definition: mapBlockDefinition,
+        legend: {
+          position: 'top',
+          items: [],
+        },
+        options: {
+          alt: '',
+          height: 300,
+          subtitle: '',
+          title: 'Chart title',
+          titleType: 'alternative',
+          boundaryLevel: 1,
+        },
+      };
```

Comment:
As above - don't need this. Just check the specific parts of state we're interested in.